File: templates/nodejs-service/service/communication/envelopedMessage.js
const { v4: uuid } = require("uuid");

class EnvelopedMessage {

    constructor(jsonMessage = null) {
        this.correlationId = "";
        this.message = {};
        this.errorMessage = "";
        if (jsonMessage) {
            const obj = typeof jsonMessage == "string" ? JSON.parse(jsonMessage) : jsonMessage;
            this.correlationId = obj.correlation_id;
            this.message = obj.message;
            this.errorMessage = obj.error;
        }
    }

    setCorrelationId(correlationId = null) {
        if (correlationId) {
            this.correlationId = correlationId;
            return this;
        }
        this.correlationId = uuid();
        return this;
    }

    setMessage(message) {
        this.message = message;
        return this;
    }

    toJson() {
        return JSON.stringify({
            correlation_id: this.correlationId,
            message: this.message,
            error: this.errorMessage,
        });
    }

}

module.exports = {
    EnvelopedMessage
};

File: templates/python-service/service/transport/envelopedMessage.py
from __future__ import annotations
from .interfaces import Message
from typing import Dict, Type, TypeVar
import uuid
import json


class EnvelopedMessage:

    def __init__(self, json_enveloped_message: str = None):
        self.correlation_id: str = ""
        self.message: Message = {}
        self.error_message: str = ""
        if isinstance(json_enveloped_message, bytes) or isinstance(json_enveloped_message, str):
            obj = json.loads(json_enveloped_message)
            self.correlation_id = obj["correlation_id"]
            self.message = obj["message"]
            self.error_message = obj["error_message"]

    def set_correlation_id(self, correlation_id: str = None) -> EnvelopedMessage:
        self.correlation_id = correlation_id if isinstance(
            correlation_id, str) else str(uuid.uuid4())
        return self

    def set_message(self, message: Message) -> EnvelopedMessage:
        self.message = message
        return self

    def to_json(self) -> str:
        data = {
            "correlation_id": self.correlation_id,
            "message": self.message,
            "error_message": self.error_message
        }
        return json.dumps(data)
